File: gdx2d-demoDesktop/src/main/java/ch/hevs/gdx2d/demos/menus/swing/DemoSwingIntegration.java
package ch.hevs.gdx2d.demos.menus.swing;


import ch.hevs.gdx2d.demos.physics.DemoSimplePhysics;
import ch.hevs.gdx2d.desktop.Game2D;
import ch.hevs.gdx2d.desktop.PortableApplication;
import com.badlogic.gdx.backends.lwjgl.LwjglAWTCanvas;
import ch.hevs.gdx2d.demos.simple.DemoCircles;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.ArrayList;

@SuppressWarnings("serial")
public class DemoSwingIntegration extends JFrame {

	int current = 0;

	ArrayList<LwjglAWTCanvas> canvasList = new ArrayList<LwjglAWTCanvas>();

	public DemoSwingIntegration() {
		setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
		this.setTitle("Embedding a gdx2demo into Swing components");
		this.setResizable(false);

		final Container container = getContentPane();
		container.setLayout(new FlowLayout());

		PortableApplication.CreateLwjglApplication = false; // TODO: configuration this way is ugly...

		JPanel jp1 = new JPanel();
		jp1.setLayout(new FlowLayout());
		JButton jb1 = new JButton("Change demo");
		jp1.add(jb1);
		container.add(jp1);

		LwjglAWTCanvas canvas = new LwjglAWTCanvas(new Game2D(new DemoSimplePhysics()));
		canvas.getCanvas().setSize(400, 400);
		container.add(canvas.getCanvas());
		canvasList.add(canvas);

		jb1.addActionListener(new ActionListener() {
			@Override
			public void actionPerformed(ActionEvent actionEvent) {
				canvasList.get(0).stop();

				LwjglAWTCanvas canvas;
				if ((current % 2) == 1)
					canvas = new LwjglAWTCanvas(new Game2D(new DemoSimplePhysics()));
				else
					canvas = new LwjglAWTCanvas(new Game2D(new DemoCircles()));

				current++;

				canvasList.clear();
				canvasList.add(0, canvas);

				container.remove(1);
				canvas.getCanvas().setSize(400, 400);
				container.add(canvas.getCanvas());
				pack();
			}
		});

		setSize(600, 500);
		setLocationRelativeTo(null);
		setVisible(true);
		pack();
	}

	public static void main(String[] args) {
		SwingUtilities.invokeLater(new Runnable() {
			@Override
			public void run() {
				new DemoSwingIntegration();
			}
		});
	}

}

File: gdx2d-demoDesktop/src/main/kotlin/ch/hevs/gdx2d/demos/menus/swing/DemoSwingIntegration.kt
package ch.hevs.gdx2d.demos.menus.swing
/*

import ch.hevs.gdx2d.demos.physics.DemoSimplePhysics
import ch.hevs.gdx2d.desktop.Game2D
import ch.hevs.gdx2d.desktop.PortableApplication
import com.badlogic.gdx.backends.lwjgl.LwjglAWTCanvas
import ch.hevs.gdx2d.demos.simple.DemoCircles

import javax.swing.*
import java.awt.*
import java.util.ArrayList

class DemoSwingIntegration : JFrame() {

    var current = 0

    var canvasList = ArrayList<Lwjgl3AWTCanvas>()

    init {
        defaultCloseOperation = JFrame.DISPOSE_ON_CLOSE
        this.title = "Embedding a gdx2demo into Swing components"
        this.isResizable = false

        val container = contentPane
        container.layout = FlowLayout()

        PortableApplication.CreateLwjglApplication = false // TODO: configuration this way is ugly...

        val jp1 = JPanel()
        jp1.layout = FlowLayout()
        val jb1 = JButton("Change demo")
        jp1.add(jb1)
        container.add(jp1)

        val canvas = LwjglAWTCanvas(Game2D(DemoSimplePhysics()))
        canvas.canvas.setSize(400, 400)
        container.add(canvas.canvas)
        canvasList.add(canvas)

        jb1.addActionListener {
            canvasList[0].stop()

            val newCanvas = LwjglAWTCanvas(Game2D(if (current % 2 == 1) DemoSimplePhysics() else DemoCircles()))

            current++

            canvasList.clear()
            canvasList.add(0, newCanvas)

            container.remove(1)
            newCanvas.canvas.setSize(400, 400)
            container.add(newCanvas.canvas)
            pack()
        }

        setSize(600, 500)
        setLocationRelativeTo(null)
        isVisible = true
        pack()
    }
}

fun main(args: Array<String>) {
  SwingUtilities.invokeLater { DemoSwingIntegration() }
}
 */
